zerver/lib/message.py

```diff
@@ -1507,16 +1506,14 @@ def stream_wildcard_mention_allowed(sender: UserProfile, stream: Stream, realm:
     return can_mention_many_users(sender)
 
 
-def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: list[int]) -> None:
-    user_groups = NamedUserGroup.objects.filter(id__in=user_group_ids).select_related(
-        "can_mention_group", "can_mention_group__named_user_group"
-    )
-
-    for group in user_groups:
-        if not sender_can_mention_group(sender, group):
+def check_user_group_mention_allowed(
+    mentioned_group_ids: set[int], mention_data: MentionData
+) -> None:
+    for group_id in mentioned_group_ids:
+        if group_id not in mention_data.allowed_mention_group_ids:
             raise JsonableError(
                 _("You are not allowed to mention user group '{user_group_name}'.").format(
-                    user_group_name=group.name
+                    user_group_name=mention_data.get_user_group_name(group_id)
                 )
             )
 
```

Comment:
Should this nclude all the non-allowed gorups ? instead of just raising `JsonableError` for the first found non-allowed group